src/cart/crt_swim.h

```diff
@@ -15,8 +15,7 @@
 #include "swim/swim_internal.h"
 
 #define CRT_SWIM_FLUSH_ATTEMPTS		100
-#define CRT_SWIM_PROGRESS_TIMEOUT	0	/* minimal progressing time */
-#define CRT_DEFAULT_PROGRESS_CTX_IDX	0
+#define CRT_DEFAULT_PROGRESS_CTX_IDX	1
```

Comment:
instead of this i was expecting something like a new option to crt_init() to override the default ctx id to use for swim. any reason why we dont allow specification of swim ctx id at startup and instead hardcode it like this? 
my understanding was that this was meant as a 'default' value if no other was specified.

Reply:
There are many places where this number is hardcoded and cannot be configurable because of it will significantly complicate the code. I can remove _DEFAULT_ from definition name if you like.

Reply:
can you explain more on this? i only see 2 places in code where CRT_DEFAULT_PROGRESS_CTX_IDX is being used:
   1    277  cart/crt_context.c <<<unknown>>>
                 ctx->cc_idx == CRT_DEFAULT_PROGRESS_CTX_IDX) {
   2    279  cart/crt_context.c <<<unknown>>>
             rc = crt_swim_init(CRT_DEFAULT_PROGRESS_CTX_IDX);

Seems both of those could just use cg_data.swim_ctx  that would be set based on init param easily.
Am I missing something?

Reply:
This IDX is used in many other places inside DAOS. So, I can pass is through param but cannot change it without bad consequence from DAOS side. SWIM itself independent from CTX_IDX and can use any but DAOS should know it in ahead to properly calculate and start xstreams, correct scheduling with set appropriate TAG.